src/test/java/com/exasol/errorcodecrawlermavenplugin/ErrorCodeCrawlerMojoIT.java

```diff
@@ -216,6 +253,59 @@ static class ITVerifier {
                 throw new IllegalStateException("Failed to create test projects src folder.");
             }
             withConfiguration("testProject/" + CONFIG_NAME);
+            if (withSubProject) {
+                withSubProject();
+            }
+        }
+
+        ITVerifier withSubProject() throws IOException {
+            this.subProjectDir = this.projectDir.resolve("sub-project");
+            if (!(this.subProjectDir.toFile().mkdirs())) {
+                throw new IllegalStateException("Failed to create sub-project folder");
+            }
```

Comment:
When you use [Files.createDirectories()](https://docs.oracle.com/en%2Fjava%2Fjavase%2F11%2Fdocs%2Fapi%2F%2F/java.base/java/nio/file/Files.html#createDirectories(java.nio.file.Path,java.nio.file.attribute.FileAttribute...))
* you don't need to check the return value
* you can directly use `Path` and don't need to convert it to `File`